cedar-policy-core/src/evaluator.rs

```diff
@@ -3777,6 +3648,16 @@ pub mod test {
         );
     }
 
+    fn assert_restricted_expression_error(v: Result<PartialValue>) {
+        match v {
+            Err(e) => assert!(matches!(
```

Comment:
`assert_matches!` gives nicer errors than `assert!(matches!(` which just will say "hey this was false and expected true"

Reply:
`assert_matches` is a nightly feature, so to enable it we'd have to add `#![feature(assert_matches)]` to the crate -- are we ok with adding this crate-level attribute?